File: mobile/lib/shared/models/sticker.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'sticker.freezed.dart';
part 'sticker.g.dart';

@freezed
class StickerPack with _$StickerPack {
  const factory StickerPack({
    required String id,
    required String name,
    required String author,
    String? description,
    required String coverUrl,
    @Default(false) bool isOfficial,
    @Default(false) bool isAnimated,
    @Default(0) int price,
    @Default(0) int downloads,
    required DateTime createdAt,
    required DateTime updatedAt,
    @Default([]) List<Sticker> stickers,
  }) = _StickerPack;

  factory StickerPack.fromJson(Map<String, dynamic> json) =>
      _$StickerPackFromJson(json);
}

@freezed
class Sticker with _$Sticker {
  const factory Sticker({
    required String id,
    required String packId,
    required String emoji,
    required String imageUrl,
    required int position,
    required DateTime createdAt,
  }) = _Sticker;

  factory Sticker.fromJson(Map<String, dynamic> json) => _$StickerFromJson(json);
}

@freezed
class UserStickerPack with _$UserStickerPack {
  const factory UserStickerPack({
    required String id,
    required String userId,
    required String packId,
    required int position,
    required DateTime createdAt,
    StickerPack? pack,
  }) = _UserStickerPack;

  factory UserStickerPack.fromJson(Map<String, dynamic> json) =>
      _$UserStickerPackFromJson(json);
}

File: backend-rs/src/models/sticker.rs
use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};
use sqlx::FromRow;
use uuid::Uuid;

#[derive(Debug, Clone, Serialize, Deserialize, FromRow)]
pub struct StickerPack {
    pub id: Uuid,
    pub name: String,
    pub author: String,
    pub description: Option<String>,
    pub cover_url: Option<String>,
    pub is_official: bool,
    pub is_animated: bool,
    pub price: i32,
    pub downloads: i64,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
}

#[derive(Debug, Clone, Serialize, Deserialize, FromRow)]
pub struct Sticker {
    pub id: Uuid,
    pub pack_id: Uuid,
    pub emoji: String,
    pub image_url: String,
    pub position: i32,
    pub created_at: DateTime<Utc>,
}

#[derive(Debug, Clone, Serialize, Deserialize, FromRow)]
pub struct UserStickerPack {
    pub id: Uuid,
    pub user_id: Uuid,
    pub pack_id: Uuid,
    pub position: i32,
    pub created_at: DateTime<Utc>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct StickerPackWithStickers {
    #[serde(flatten)]
    pub pack: StickerPack,
    pub stickers: Vec<Sticker>,
}
